### ISSUE TITLE
HTTP requests are blocked

### ISSUE THREAD

**Issue by radtrentasei** [TurboWarp/extensions#642]
Hello team,

I am working on an extension that consume REST API exposed via HTTP. 

From desktop version v1.8.1 I am unable to consume these APIs (I am getting net::ERR_BLOCKED_BY_CLIENT error in the console but NO WARNING like I was getting in the online version even before 1.8.1). It did work on the v1.7.1. It never worked on the online editor version.

I am working on enabling HTTPS on the target server, however, this might require some time sadly for me. 

I was wondering if there is any workaround for this. Thanks!

Apologise if this question has been asked already.

**Comment by GarboMuffin** [comment]
Your website should be setting appropriate CORS headers

If that's not possible, there is an option in the desktop app settings to disable CORS checks

**Comment by GarboMuffin** [comment]
The correct fix again is to set CORS headers, which will also make it work in the online editor

If you can't do that, press (?) > Desktop settings > Allow extensions to access any website

**Comment by radtrentasei** [comment]
Thank you for the help.

Indeed disabling the CORS checks the extension works with HTTP.

In the meantime I have managed to add HTTPS to the API provider and now I am facing a new challenge.

With the self-signed certificate I've noted that the desktop app refuses to establish connection while the online editor at first refuses, if then I visit the URL with the browser and trust the certificate and go back to the online editor it complains about the cors check with this error message 

```
Acess to fetch at 'https://10.10.20.50/getconfig' from origin 'https://turbowarp.org' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

The question now is, is there a way to enable self-signed certificates both on the online and the desktop? Maybe something I can do on the code itself?

Thanks

**Comment by GarboMuffin** [comment]
If the API is only on your LAN then HTTPS is probably unnecessary and, as you've experienced, going to cause all sorts of problems (if you train yourself to manually trust the certificate each time you get an error, it also gains you precisely nothing in security)

Ignoring certificate errors is such a dangerous edge case that I won't be providing official support for that. You could probably install that certificate into your system's global certificate storage, or find some Electron argument that lets you disable certificate checking

**Comment by GarboMuffin** [comment]
I forgot that HTTPS would be needed to make your API work in the website editor. The real path forward there is probably a real actual HTTPS certificate for some domain. There are sites that give you free (sub)domains and places where you can get free wildcard certificates so your actual website can still be only on your LAN

**Comment by GarboMuffin** [comment]
Pretty sure there are also some Chrome/Edge flags to make it treat a given IP address as always secure even if it's definitely insecure. I don't think there's an elegant solution to your problem.

**Comment by radtrentasei** [comment]
Thanks a lot for the follow-up.

I am investigating the path of getting a certificate signed from a public CA to disable all the warnings.
Security is not really a concern here since the API backend (that supports only HTTP) is accessible via a VPN that points to a close sandbox. Anyway if I manage to the proper certificate it's better.

However now I am a little bit lost with the HTTPS 'tentative'. I am running this from Chrome, I've manually trusted the self-signed certificate from the browser but I am getting the error message I've posted in the previous message.

I've tried to run the fetch with the 'no-cors' but then the client complains again saying that the redirect is not follow (BUT IT IS FOLLOW)

Consider that I've added a proxy (ngnix) in front of the API broker that enables HTTPS
